pkg/graph/resolvers/model_plan.go

```diff
@@ -279,3 +282,118 @@ func ModelPlanNameHistory(logger *zap.Logger, modelPlanID uuid.UUID, sortDir mod
 
 	return nameHistory, nil
 }
+
+// ModelPlanShare implements resolver logic to share a model plan with a list of emails
+func ModelPlanShare(
+	ctx context.Context,
+	logger *zap.Logger,
+	store *storage.Store,
+	principal authentication.Principal,
+	emailService oddmail.EmailService,
+	emailTemplateService email.TemplateService,
+	modelPlanID uuid.UUID,
+	viewFilter *models.ModelViewFilter,
+	receiverEmails []string,
+	optionalMessage *string,
+) (bool, error) {
+	modelPlan, err := store.ModelPlanGetByID(logger, modelPlanID)
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			logger.Warn("No model plan found for the given modelPlanID",
+				zap.String("modelPlanID", modelPlanID.String()))
+			return false, fmt.Errorf("no model plan found for the given modelPlanID: %w", err)
+		}
```

Comment:
I don't think we need to have specific error handling for `ErrNoRows` here - we should probably just keep this simple, since we're returning an error either way, we'll be able to see what the source of the issue was without having to check it here

Reply:
The thinking here was to provide more expressive errors so that we could understand the context of the 'pq: no rows' error. Please let me know if you still want this removed and I will remove it.

Reply:
I think we can do that, but if we want to wrap the errors I'd say doing it in the `store` method might make more sense rather than having to do it in each resolver that calls it, that way we can just forward the error up the call stack in the resolvers

Reply:
Good point.